src/Chapter3.hs

```diff
@@ -1007,7 +1126,22 @@ properties using typeclasses, but they are different data types in the end.
 Implement data types and typeclasses, describing such a battle between two
 contestants, and write a function that decides the outcome of a fight!
 -}
-
+data Warrior = Warrior
+  { warriorHealth :: Health,
+    warriorAttack :: Attack,
+    warriorDefence :: Defense,
+    warriorActions :: [WarriorAction]
+  }
+
+data WarriorAction = DrinkPotion | BuffSpell
+
+data Creature = Creature
+  { creatureHealth :: Health,
+    creatureAttack :: Attack,
+    creatureAction :: [CreatureAction]
+  }
+
+data CreatureAction = JustRun
```

Comment:
Nice data types created here 🥇  Now you need to create `Fighter` typeclass as well.
Here are a few hints on the task if you would wish to come back to it later 🙂 

Since each data type (`Monster` and `Knight`) can be fighters, I can suggest creating a typeclass of the following shape:

```haskell
class Fighter a where
    ... you need to think of possible methods here ...
```

So you can them implement a recursive function using this typeclass:

```haskell

data FightResult a b
    = FirstWins a
    | SecondWins b

fight :: (Fighter a, Fighter b) => a -> b -> FightResult a b
```

I hope this clarifies the task a bit more and gives some ideas on how to proceed with it :slightly_smiling_face:

Reply:
Thank you for that comment. I might come back to this exercise when I practice some more, but for now I'll merge this pull request as it is.